### DISCUSSION TITLE
TCP Listener Message Recovery???

### DISCUSSION THREAD

**Discussion by elemle** [nextgenhealthcare/connect#4744] [Q&A]
In the TCP Listener and MLLP should I be using the Message Recovery option to resolve this issue we have....
When our Hosting servers temporarily turn off firewalls or something similar making clients unable to connect.
I would think that mirth connect would just wait and not be busy because no new HL7 QBP messages are coming in.
But, what actually happens is that the CPU of the Linux Server that Mirth Connect is running on goes sky high (maxed out) 
when this happens.  Is this because mirth connect code freaks out when it can't respond through the network anymore?  Would using this setting [Respond on New Connection-->Message Recovery] below help resolve this issue?  What is mirth connect code doing?  Instantiating more and more threads of something?  (Mirth Connect Server 3.10.0 Built on November 5, 2020 Java version: 1.8.0_181)
Any info will be appreciated!
![TCP](https://user-images.githubusercontent.com/81604881/134379517-179e2fc7-8b29-4ab8-b17a-22de5b4b9ab5.png)

**Comment by pacmano1** [comment]
I don't think I have ever used "respond on new connection" since the other side needs to have a listener configured for that.  

I am not clear on when that happens with your set up - are you saying mid-ACK (or other response since it is an inbound query)  the firewall breaks the connection and mirth goes nuts?

**Reply by elemle** [reply]
We have many threads running, but yes, when they can't respond back to the QBP request (no network path) something starts sucking up all the CPU cycles (mirth maybe, postgreSQL maybe, linux OS?)

**Comment by pacmano1** [comment]
LOL - I think you will have to tell us what process(es) is doing that via interactive access on your linux box.

**Reply by elemle** [reply]
I am more interested in what this setting is really used for, what problem does it resolve?  The Hint does not tell me when I would want to use it.

**Reply by tonygermano** [reply]
Rather than responding on the same connection, the ACK will be sent on a separate connection. Like @pacmano1 mentioned in another thread, the other end has to be configured for this, as well. If the thing you are connecting to doesn't request it, then you don't need it.
